prov/sockets/src/sock_epoll.c

```diff
@@ -61,10 +61,14 @@
 #ifdef HAVE_EPOLL
 int sock_epoll_create(struct sock_epoll_set *set, int size)
 {
+	int ret;
 	set->size = size;
 	set->used = 0;
 	set->events = calloc(size, sizeof(struct epoll_event));
-	return set->fd = epoll_create(size);
+	ret = set->fd = epoll_create(size);
+	if (ret < 0)
+		free(set->events);
```

Comment:
This has modified sock_epoll_set.  set->events is freed here in the error case.  Is set->events also freed from another function?  I didn't check, but it seems likely, unless we have a memory leak somewhere else.  If so, we have a double free.